common/src/main/java/com/microsoft/identity/common/internal/providers/oauth2/AuthorizationActivityFactory.java

```diff
@@ -0,0 +1,126 @@
+     * @param authorizationAgent         The means by which authorization should be performed (EMBEDDED, WEBVIEW) NOTE: This should move to library configuration
+     * @param webViewZoomEnabled         This parameter is specific to embedded and controls whether webview zoom is enabled... NOTE: Needs refactoring
+     * @param webViewZoomControlsEnabled This parameter is specific to embedded and controls whether webview zoom controls are enabled... NOTE: Needs refactoring
+     * @return An android Intent which will be used by Android to create an AuthorizationActivity
+     */
+    public static Intent getAuthorizationActivityIntent(final Context context,
+                                                        final Intent authIntent,
+                                                        final String requestUrl,
+                                                        final String redirectUri,
+                                                        final HashMap<String, String> requestHeaders,
+                                                        final AuthorizationAgent authorizationAgent,
+                                                        final boolean webViewZoomEnabled,
+                                                        final boolean webViewZoomControlsEnabled) {
+        Intent intent;
+        final LibraryConfiguration libraryConfig = LibraryConfiguration.getInstance();
+        if (ProcessUtil.isBrokerProcess(context)) {
+            intent = new Intent(context, BrokerAuthorizationActivity.class);
+        } else if (libraryConfig.isAuthorizationInCurrentTask()) {
+                intent = new Intent(context, CurrentTaskAuthorizationActivity.class);
+        } else {
+                intent = new Intent(context, AuthorizationActivity.class);
+        }
+
+        intent.putExtra(AUTH_INTENT, authIntent);
+        intent.putExtra(REQUEST_URL, requestUrl);
+        intent.putExtra(REDIRECT_URI, redirectUri);
+        intent.putExtra(REQUEST_HEADERS, requestHeaders);
+        intent.putExtra(AUTHORIZATION_AGENT, authorizationAgent);
+        intent.putExtra(WEB_VIEW_ZOOM_CONTROLS_ENABLED, webViewZoomControlsEnabled);
+        intent.putExtra(WEB_VIEW_ZOOM_ENABLED, webViewZoomEnabled);
+        intent.putExtra(DiagnosticContext.CORRELATION_ID, DiagnosticContext.getRequestContext().get(DiagnosticContext.CORRELATION_ID));
+        return intent;
+    }
+
+    /**
+     * Returns the correct authorization fragment for local (non-broker) authorization flows.
+     * Fragments include:
+     * {@link WebViewAuthorizationFragment}
+     * {@link BrowserAuthorizationFragment}
+     * {@link CurrentTaskBrowserAuthorizationFragment}
+     *
+     * @param intent
+     * @return returns an Fragment that's used as to authorize a token request.
+     */
+    public static Fragment getAuthorizationFragmentFromStartIntent(@NonNull final Intent intent) {
```

Comment:
I'm wondering if this factory method is suitable for what we want to do here.

If I'm using CurrentTaskAuthorizationActivity,  I would expect to only get Fragment of the same category, but this method, if a bug is introduced in the future, could return a singleTask fragment.

We might want to let the inherited class(es) decide which one to pick.

Reply:
Yes.  I'll file a work item to make this an abstract factory in a future release.